src/ex6/src/client/src/components/TodoItem.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { FaTrash } from 'react-icons/fa';
+import PropTypes from "prop-types";
+import RemoteItemManager from '../remote-item-manager.js';
+
+function TodoItem({ todos, fetchTodos }) {
+    const remoteItemManager = new RemoteItemManager();
+    
+    const alertMessage = (event) => {
+        if ( event.target.className !== 'deleteButton' && event.target.className !== 'doneCheckbox') {
+            alert(event.target.innerText);
+        }
+    }
+    
+    const doneTodo = async (id) => {
+        await remoteItemManager.setDone(id);
+        fetchTodos();
+    }
+
+    const unDoneTodo = async (id) => {
+        await remoteItemManager.setUnDone(id);
+        fetchTodos();
+    }
+
+    const removeTodo = async (id) => {
+        await remoteItemManager.deleteItem(id);
+        fetchTodos();
+    }
+
+    return todos.map((todo, index) => (
+        <div className={todo.status ? 'li_done' : 'li'} key={index}>
+               <div key={todo.id} onClick={alertMessage} id="task-list-item">
+                <input type="checkbox" onClick={() => todo.status ? unDoneTodo(todo.id) : doneTodo(todo.id)} checked={todo.status ? 'checked' : ''} className='doneCheckbox' />
```

Comment:
this way you creats a new ref of the function for each render
you can wrap with function the toggle the status of the id 
```suggestion
                <input type="checkbox" onClick={toggleStatus(todo.status, todoId)} checked={todo.status ? 'checked' : ''} className='doneCheckbox' />
```